lib/src/Uncertainty/Algorithm/Simulation/NAIS.cxx

```diff
@@ -92,17 +92,17 @@ Distribution NAIS::computeAuxiliaryDistribution(const Sample & sample,
   Collection<Distribution> margins(dimensionSample);
 
   // Computation of auxiliary distribution using ot.Mixture
-  for (UnsignedInteger k = 0; k < dimensionSample ; ++k)
+  const UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
+  Collection<Distribution> collectionOfDistribution(numberOfSample);
+  for (UnsignedInteger i = 0; i < numberOfSample ; ++i)
   {
-    const UnsignedInteger numberOfSample = getMaximumOuterSampling() * getBlockSize();
-    Collection<Distribution> collectionOfDistribution(numberOfSample);
-    for (UnsignedInteger i = 0; i < numberOfSample ; ++i)
-    {
-      collectionOfDistribution[i] = Normal(sample(i, k), silverman[k]);
-    }
-    margins[k] = Mixture(collectionOfDistribution, weights);
-  } // for k
-  return ComposedDistribution(margins);
+    const Point  mean_normal =  sample[i];
+    
+    collectionOfDistribution[i] = Normal(mean_normal, silverman);
+  }
+  Distribution auxiliaryDistribution = Mixture(collectionOfDistribution, weights);
```

Comment:
Same ramrk here, plus const correctness. If `auxiliaryDistribution` is not modified afterward, use:
```
const Distribution auxiliaryDistribution(Mixture(collectionOfDistribution, weights));
```
or even better:
```
const Mixture auxiliaryDistribution(collectionOfDistribution, weights);
```

Reply:
Thank you, modification has been done accordingly